src/fty_config_manager.cc

```diff
@@ -0,0 +1,448 @@
+        } catch (...)
+        {
+            log_error("Unexpected error: unknown");
+        }
+    }
+    
+    /**
+     * Set response before to sent it.
+     * @param msg
+     * @param responseDto
+     * @param configQuery
+     */
+    void ConfigurationManager::setSaveResponse (const std::map<std::string, cxxtools::SerializationInfo>& configSiList, dto::config::ConfigResponseDto& respDto)
+    {
+        // Array si
+        cxxtools::SerializationInfo jsonResp;
+        jsonResp.setCategory(cxxtools::SerializationInfo::Category::Array);
+
+        for(auto const& configSi: configSiList)
+        {
+            cxxtools::SerializationInfo si;
+            si.setCategory(cxxtools::SerializationInfo::Category::Object);
+            // Feature si
+            cxxtools::SerializationInfo siFeature;
+            // Content si
+            cxxtools::SerializationInfo siTemp;
+            siTemp.addMember(SRR_VERSION) <<= ACTIVE_VERSION;
+            cxxtools::SerializationInfo& siData = siTemp.addMember(DATA_MEMBER);
+            siData <<= configSi.second;
+            siData.setName(DATA_MEMBER);
+            // Add si version + data in si feature
+            siFeature <<= siTemp;
+            siFeature.setName(configSi.first);
+            // Add the feature in the main si
+            si.addMember(configSi.first) <<= siFeature;
+            // Put in the array
+            jsonResp.addMember(configSi.first) <<= si;
+        }
+        // Serialize the response
+        respDto.data = JSON::writeToString (jsonResp, false);
+        respDto.status = STATUS_SUCCESS;
+    }
+    
+    /**
+     * Send response on message bus.
+     * @param msg
+     * @param responseDto
+     * @param configQuery
+     */
+    void ConfigurationManager::sendResponse(const messagebus::Message& msg, const messagebus::UserData& userData, const std::string& subject)
+    {
+        try
+        {
+            messagebus::Message resp;
+            resp.userData() = userData;
+            resp.metaData().emplace(messagebus::Message::SUBJECT, subject);
+            resp.metaData().emplace(messagebus::Message::FROM, m_parameters.at(AGENT_NAME_KEY));
+            resp.metaData().emplace(messagebus::Message::TO, msg.metaData().find(messagebus::Message::FROM)->second);
+            resp.metaData().emplace(messagebus::Message::COORELATION_ID, msg.metaData().find(messagebus::Message::COORELATION_ID)->second);
+            m_msgBus->sendReply(msg.metaData().find(messagebus::Message::REPLY_TO)->second, resp);
+        }
+        catch (messagebus::MessageBusException& ex)
+        {
+            log_error("Message bus error: %s", ex.what());
+        } catch (...)
+        {
+            log_error("Unexpected error: unknown");
+        }
+    }
+
+    /**
+     * Set a configuration.
+     * @param si
+     * @param path
+     * @return 
+     */
+    int ConfigurationManager::setConfiguration(cxxtools::SerializationInfo* si, const std::string& path)
```

Comment:
Change prototype to pass a const reference instead of a pointer to a `cxxtools::SerializationInfo` object.